lib/charms/zookeeper/v0/client.py

```diff
@@ -425,6 +425,24 @@ def delete_znode_leader(self, path: str) -> None:
         ) as zk:
             zk.delete_znode(path=path)
 
+    def get_version(self) -> str:
```

Comment:
**todo:** This should take an optional argument for `host`, defaulting to `self.leader`. In case ZooKeeper version differs between leader/non-leader units during an upgrade.

We also should account for scenarios where the result fails by returning `None` (although realistically this would never happen).

```python
def get_version(self, host: Optional[str] = None) -> Optional[str]:
    server_host = host or self.leader
```

Reply:
In other places like property `self.config_version` only talks to the leader. Isn't this the same situation?

Reply:
Arguably yeah it is the same. I think our use-case is different here though. Maybe we can add it to `ZooKeeperClient`, which connects to a specific unit, then update the docs to make it 'usable'.

What do you think?

Reply:
I honestly agree with @zmraul . I understand that this does not cover the upgrade where members may have transiently different versions, but I don't think that clients should worry about this. I believe clients should only see the "ufficial version" which is what the leader is communicating